# <p align="center"> Awesome Phenotype Prediction </p> 

## Contents
- [Introduction](#introduction)
- [Papers](#papers)


# Introduction
## <p align="center"> What is phenotype prediction? </p>
 
Phenotype Prediction consists in predicting an organism’s phenotype from the knowledge of its genotype and its environment. 

For example, a human disease is a phenotype, and understanding its relation to genotype and environment is a central problem in medicine. On the other hand, crop yield and drought resistance are phenotypes, and if we are going to be able to continue to feed the world’s growing population, it is essential to better predict crop phenotypes from knowledge of their genotypes and environment. 

Our ability to predict phenotype is being revolutionised by advances in DNA sequencing technology. The cost of sequencing the DNA is decreasing rapidly, which means that it is now often low enough that in a single investigation many organisms may be genotyped, which opens up the possibility of using machine learning to learn predictive relationships between an organism’s genotype, environment, and phenotype. Such studies are often called genome-wide association studies (GWAS).
<br />



</br>

<p align="center">
  <img width="650" height="500" src="https://user-images.githubusercontent.com/90758727/162408525-5cb042f9-e708-4e55-87be-8c99e865f808.png">
</p>

<div style="margin-top: 0px;">
<p align="center">
<em>Manhattan plot</em>
</p>
</div>


<p align="center">
<a href="https://www.genome.gov/genetics-glossary/Genome-Wide-Association-Studies" align= "center">Reference</a>
</p>
<br />



</br>

## <p align="center"> Challenges in the prediction </p>
### <p align = 'center'> Why an individual's phenotype cannot always be predicted from their genome and the environment they experience? </p>

One of the main things scientists ask ourselves in phenotype prediction is to what extent is the knowledge about an individual's genotype, together with information about the environment that they have experienced, sufficient to predict phenotypic variation.

In the referenced paper, they argue that, although the 'typical' phenotypic outcome of an individual's genome can be predicted, it is much more difficult to predict the actual outcome for a particular individual. One of the three main reasons for this are:
1. The outcome of mutations can be influenced by random stochastic processes. 
2. Genetic variation present in one generation can influence phenotypic traits in the next generation.
3. The environment experienced by one generation can influence phenotypic variation in the next generation. 

Taken together, they mean that, in many cases, the genotypes of individuals and the environment that they experience may not be sufficient to determine their phenotypes. This is why, in this research we want to give a different approach to phenotype prediction.

<p align="center">
<a href="https://pubmed.ncbi.nlm.nih.gov/22934970/" align= "center">Reference</a>
</p>


  
  
  

---

# Papers

  - ## Phenotype Prediction
    - **Checked**
       1. deepManReg: Predict phenotypes from multi-modal data. [[paper]](https://www.nature.com/articles/s43588-021-00185-x) [[code]](https://github.com/daifengwanglab/deepManReg)
       2. GWAS DCNN Soybean: [[paper]](https://www.frontiersin.org/articles/10.3389/fgene.2019.01091/full) [[code]](https://github.com/kateyliu/DL_gwas)
    
    
    - **Unchecked**
      1. https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-3427-8
      2. https://link.springer.com/article/10.1007/s10994-019-05848-5
      3. https://www.nature.com/articles/s41598-019-40561-2
      4. https://www.nature.com/articles/s41467-021-25893-w
      5. https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04077-9
      6. https://www.biorxiv.org/content/10.1101/2021.05.27.446033v1 
    
  - ## PRS
    - **Checked**
      1. Guide to PRS analyses: [[paper]](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7612115/) [[code]](https://choishingwan.github.io/PRS-Tutorial/) [[reference to our tutorials]](https://github.com/AI-sandbox/Getting-Started/blob/main/video_tutorial.md)
      2. Inferred relatives: Exploiting phenotypes from inferred relatives. [[paper]](https://www.nature.com/articles/s41467-020-16829-x) [[code]](https://github.com/bulik/ldsc)
      3. Non-linearities PRS. [[paper]](https://www.medrxiv.org/content/10.1101/2021.07.09.21260288v1.full)
    - **Unchecked**
      1. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6456317/
      2. https://www.sciencedirect.com/science/article/abs/pii/S0168952521001451
    
  - ## Multi-ethnicity // Cross-Ethnicity
    - **Checked**
      1. Cardiovascular disease risk prediction: [[paper]](https://www.nature.com/articles/s41746-020-00331-1)[[code]](https://github.com/atward424/ASCVD_ML/)
      2. Predict ethnicity using personal name Canada.[[paper]](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0241239) [[code]](https://github.com/kaionwong/ethnicity-ml-prediction)
    
  - ## Machine Learning
    - **Checked**
      1. Cardiovascular disease risk prediction: [[paper]](https://www.nature.com/articles/s41746-020-00331-1)[[code]](https://github.com/atward424/ASCVD_ML/)  
    - **Unchecked**
      1. https://pubmed.ncbi.nlm.nih.gov/28794054/ 

- ## Neural Networks
    - **Checked**
      
    - **Unchecked**
      1. https://www.proquest.com/openview/61637c22f25bb23a85eec1c4b157ce71/1?pq-origsite=gscholar&cbl=18750&diss=y  
